x12_test.go

```diff
@@ -144,6 +161,9 @@ IEA*1*000095071~`,
 				t.Errorf("Decode() error = %v, wantErr %v", err, tt.wantErr)
 				return
 			}
+			if err != nil {
```

Comment:
shouldn't this include a wantErr check?

Reply:
I was just looking a out following decode having an error. If err is not nil, then got is nil, which causes validate to panic.
I suppose the better option would to check if got is nil before running validate. Honestly I haven't tried to pull TestDecode apart enough to understand everything it is trying to do, so was just doing the minimal amount of work to have the test act as expected without panics.